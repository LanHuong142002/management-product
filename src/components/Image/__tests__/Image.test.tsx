import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import { Image, ImageProps } from '@components';

describe('Testing Image component', () => {
  afterEach(() => {
    cleanup();
  });

  const defaultProps = {
    url: 'image.jpg',
  } as ImageProps;

  it('Should render Image', () => {
    const { container } = render(<Image {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  it('Should call onClick event when click on image', () => {
    const handleClick = jest.fn();
    render(<Image {...defaultProps} isClickable={true} onClick={handleClick} />);

    const figure = screen.getByRole('figure');
    fireEvent.click(figure);

    expect(handleClick).toBeCalled();
    expect(handleClick).toBeCalledTimes(1);
  });

  it('Should render image with circle shape', () => {
    const handleClick = jest.fn();
    render(<Image {...defaultProps} isCircle={true} isClickable={false} />);

    const figure = screen.getByRole('figure');
    fireEvent.click(figure);

    expect(handleClick).not.toBeCalled();
  });
});
