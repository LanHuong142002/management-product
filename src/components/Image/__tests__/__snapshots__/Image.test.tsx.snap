// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Image component Should render Image 1`] = `
<div>
  <figure
    class="image-wrapper image-size-xs "
  >
    <img
      alt="image"
      class="image "
      src="image.jpg"
    />
  </figure>
</div>
`;
